polaris-react/stylelint/plugins/coverage/rules/global-disallowed-list/index.js

```diff
@@ -0,0 +1,61 @@
+const {ruleMessages, validateOptions, report} = require('stylelint').utils;
+
+const {isRegExp, isString, namespace} = require('../../utils');
+
+const ruleName = namespace('global-disallowed-list');
+
+const messages = ruleMessages(ruleName, {
+  rejected: (value) => `Unexpected disallowed value "${value}"`,
+});
+
+const rule = (primary) => {
+  return (postcssRoot, postcssResult) => {
+    const validOptions = validateOptions(postcssResult, ruleName, {
+      actual: primary,
+      possible: [isRegExp, isString],
+    });
+
+    if (!validOptions) {
+      return;
+    }
```

Comment:
Shouldn't we inform folks their config is malformed, rather the fail silently? Here is an example of how we handle misconfigurations with the `custom-properties-allowed-list` plugin: https://github.com/Shopify/polaris/blob/399556256b44c36559abda717299f2064895df3d/polaris-react/stylelint/plugins/custom-properties-allowed-list/index.js#L63-L82

Reply:
Yep yep, great idea. Added a very basic error message ⚠️ not as fancy with the options parsing but should certainly help with pointing out config issues